### PR TITLE
8308283: Build failure with GCC12 & GCC13

### PR DESCRIPTION
**Pull request by xpbob**
configure --enable-debug
```
error: infinite recursion detected [-Werror=infinite-recursion]
  193 | void VMError::reattempt_test_hit_stack_limit(outputStream* st)
```
configure
```
java.desktop/share/native/libharfbuzz/graph/../hb-ot-layout-common.hh:1161:24: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
 1161 | const LangSys& l = this+_.second.offset; 
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8308283](https://bugs.openjdk.org/browse/JDK-8308283): Build failure with GCC12 &amp; GCC13

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [1c80f579](https://git.openjdk.org/jdk/pull/14032/files/1c80f57986247f0313970888288206993963b2ad)
 * [Jie Fu](https://openjdk.org/census#jiefu) (@DamonFool - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14032/head:pull/14032` \
`$ git checkout pull/14032`

Update a local copy of the PR: \
`$ git checkout pull/14032` \
`$ git pull https://git.openjdk.org/jdk.git pull/14032/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14032`

View PR using the GUI difftool: \
`$ git pr show -t 14032`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14032.diff">https://git.openjdk.org/jdk/pull/14032.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14032#issuecomment-1551312063)

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]
Build changes look good. Someone from hotspot should also weigh in.

/reviewers 2

**Comment by prrace** [comment]
This is a duplicate of https://bugs.openjdk.org/browse/JDK-8307210
Please check for existing reports before creating a new bug.
We already plan to fix this before we officially upgrade to 13.1, until then you just disable the warning.

**Comment by DamonFool** [comment]
@xpbob , maybe, it would be better to change the JBS title with something like `Build failure with GCC12 & GCC13`?

**Comment by xpbob** [comment]
The patch here involves two different parts. I tried to change the JBS title

**Comment by xpbob** [comment]
@prrace hi
I found the error: infinite recursion detected [-Werror=infinite-recursion] message and check  the information in existing reports,There are no similar bugs.

**Comment by xpbob** [comment]
@DamonFool @prrace 
I have changed the JBS title

File: src/hotspot/share/utilities/compilerWarnings_gcc.hpp [hunk 1]
```diff
@@ -44,6 +44,11 @@
 #define PRAGMA_DANGLING_POINTER_IGNORED PRAGMA_DISABLE_GCC_WARNING("-Wdangling-pointer")
 #endif
 
+// Disable -Winfinite-recursion which is introduced in GCC 12.
+#if !defined(__clang_major__) && (__GNUC__ >= 12)
+#define PRAGMA_INFINITE_RECURSION_IGNORED PRAGMA_DISABLE_GCC_WARNING("-Winfinite-recursion")
```
Comment (DamonFool) [inline src/hotspot/share/utilities/compilerWarnings_gcc.hpp:49]: How about moving this line after Line 44?

**Comment by xpbob** [comment]
@DamonFool 
Thanks for the review,The code has been updated

**Review by DamonFool — APPROVED** [review]
LGTM
Thanks for the update.

**Pull request closed without merging**